packages/react-instantsearch-core/src/lib/useSearchState.ts

```diff
@@ -20,6 +22,8 @@ export function useSearchState<
   const indexId = searchIndex.getIndexId();
   const [uiState, setLocalUiState] = useState(() => search.getUiState());
   const indexUiState = uiState[indexId] as TUiState[string];
+  const [renderState, setRenderState] = useState(() => search.renderState);
+  const indexRenderState = renderState[indexId] || {};
```

Comment:
Is there a reason why we have this `{}` fallback for the index render state but it's not needed for the index UI state?

Reply:
getUiState uses the mounted index widgets (all) as its start value, but renderState starts with an empty object